server/src/main/java/org/apache/iotdb/db/query/fill/PreviousFill.java

```diff
@@ -59,28 +84,184 @@ public long getBeforeRange() {
+
+          for (int i = seqChunkMetadataList.size() - 1; i >= 0; i--) {
+            lastPoint = getChunkLastPoint(seqChunkMetadataList.get(i));
+            // last point of this sequence chunk is valid, quit the loop
+            if (lastPoint.getValue() != null) {
+              return lastPoint;
+            }
+          }
+        }
+      }
+    }
+
+    return lastPoint;
+  }
+
+  /**
+   * find the last TimeseriesMetadata and unpack all overlapped seq/unseq files
+   */
+  private void UnpackOverlappedUnseqFiles(long lBoundTime) throws IOException {
+    PriorityQueue<TsFileResource> unseqFileResource =
+        sortUnSeqFileResourcesInDecendingOrder(dataSource.getUnseqResources());
+
+    while (!unseqFileResource.isEmpty()) {
+      // The very end time of unseq files is smaller than lBoundTime,
+      // then skip all the rest unseq files
+      if (unseqFileResource.peek().getEndTimeMap().get(seriesPath.getDevice()) < lBoundTime) {
+        return;
+      }
+      TimeseriesMetadata timeseriesMetadata =
+          FileLoaderUtils.loadTimeSeriesMetadata(
+              unseqFileResource.poll(), seriesPath, context, timeFilter, allSensors);
+      if (timeseriesMetadata != null
+          && lBoundTime <= timeseriesMetadata.getStatistics().getEndTime()) {
+        lBoundTime = Math.max(lBoundTime, timeseriesMetadata.getStatistics().getStartTime());
+        unseqTimeseriesMetadataList.add(timeseriesMetadata);
+        break;
+      }
+    }
+
+    // unpack all overlapped unseq files and fill unseqTimeseriesMetadata list
+    while (!unseqFileResource.isEmpty()
+        && (lBoundTime <= unseqFileResource.peek().getEndTimeMap().get(seriesPath.getDevice()))) {
+      TimeseriesMetadata timeseriesMetadata =
+          FileLoaderUtils.loadTimeSeriesMetadata(
+              unseqFileResource.poll(), seriesPath, context, timeFilter, allSensors);
+      unseqTimeseriesMetadataList.add(timeseriesMetadata);
+      // current unseq timeseriesMetadata's last point is a valid result,
+      // then skip the rest unseq files
+      if (endtimeContainedByTimeFilter(timeseriesMetadata.getStatistics())) {
         break;
       }
     }
+  }
+
+  private TimeValuePair getChunkLastPoint(ChunkMetadata chunkMetaData) throws IOException {
+    TimeValuePair lastPoint = new TimeValuePair(0, null);
+    Statistics chunkStatistics = chunkMetaData.getStatistics();
+
+    if (endtimeContainedByTimeFilter(chunkStatistics)) {
+      return constructLastPair(
+          chunkStatistics.getEndTime(), chunkStatistics.getLastValue(), dataType);
+    }
+    List<IPageReader> pageReaders = FileLoaderUtils.loadPageReaderList(chunkMetaData, timeFilter);
+    for (int i = pageReaders.size() - 1; i >= 0; i--) {
+      IPageReader pageReader = pageReaders.get(i);
+      Statistics pageStatistics = pageReader.getStatistics();
+      if (!timeFilter.satisfy(pageStatistics)) {
+        continue;
+      }
```

Comment:
no need to check, pageReaders got from chunkmetadata should satisfy the timeFilter already